### ISSUE TITLE
Disentangle gridding for dirty image from gridding for PyTorch export

### ISSUE THREAD

**Issue by iancze** [MPoL-dev/MPoL#36]
**Is your feature request related to a problem or opportunity? Please describe.**

Right now, the [Gridder](https://github.com/MPoL-dev/MPoL/blob/main/src/mpol/gridding.py#L44) uses one routine (`grid_visibilities`) for preparing a Fourier Grid with visibilities for both dirty imaging (numpy) and for export to PyTorch as a GriddedDataset for RML imaging. 

This is possible because a Fourier grid prepared for 'uniform'ly-weighted imaging is the correct statistical interpretation for averaged visibilities.

However, the fact that this routine is overloaded makes things confusing and (probably) error prone. For example, if someone called 

```
gridder.grid_visibilities(weighting="uniform")
gridder.grid_visibilities(weighting="natural")
dataset = gridder.to_pytorch_dataset()
```
they might still be able to export the dataset to PyTorch even though the statistical interpretation of the weights would be incorrect for RML imaging.
 
**Describe the solution you'd like**
There are two workflows that we expect a user would encounter, and we want them to be decoupled each other and the internal state of the grid.

1. create a dirty image (along w/ dirty beam) to inspect the data. Cell-averaging choices include uniform, robust, natural. This would be like a "dirty_image" routine that returns the image and beam. Also stores them to the object.
2. export a GriddedDataset object to run with RML. Because this is a condensed (averaged) statistical representation of the raw data, there's only one right choice here. That is a weighted mean, which just happens to be equivalent to "uniform" imaging. 

Create a 

* `_grid_visibilities` which just carries out the relevant averaging from loose visibilities
* `_grid_weights` which just grids weights
* `dirty_image(weighting)` routine calls `_grid_visibilities`, with the weighting, stores vis, then calcs beams for export.
* `to_pytorch` calls `_grid_visibilites` w/ uniform, `grid_weights`, and then creates the `GriddedDataset`
* while we're here, we probably want to think about adding an `_estimate_cell_variance(gridded_weights, loose visibilities)` routine, or something like it, to address #37 during pytorch export. it could be called in between`_grid_weights` and `_grid_visibilities` and and raise a warning if the variance is higher than the weights would suggest.

Basically, we don't want the user to have to call `_grid_visibilites` directly. They approach the object really only needing to call `get_dirty_image` or `to_pytorch_dataset`.

There might be some code duplication with this solution, but the user interface will be much nicer.

What tests are needed to verify this functionality?

**Comment by briannazawadzki** [comment]
[I did some planning in a text file](https://github.com/MPoL-dev/MPoL/blob/weights/issue36_plan.txt) on the weights branch.

It's not a full new version of gridder.py yet, only `get_dirty_image`, `to_pytorch_dataset`, `_grid_visibilities`, and `_grid_weights`. Rather than going for full pseudo code I ended up taking what we already had and just modifying it directly. @iancze let me know if you think this is heading in the right direction.

**Comment by iancze** [comment]
Your suggested design sounds great! I would go ahead and implement these changes directly in the gridder.py file. Then, run the tests, see what fails, and edit/write new tests as necessary.

Feel free to push new changes to this branch when you'd like feedback. I created a draft pull request (#50) that will allow us to comment directly on the committed changes as we go. Then, when everything's complete, we can change the status from "draft" and merge the branch directly into main.
